src/storage/src/generated/gapic/transport.rs

```diff
@@ -381,7 +476,24 @@ impl super::stub::Storage for Storage {
         };
         let path =
             http::uri::PathAndQuery::from_static("/google.storage.v2.Storage/CancelResumableWrite");
-        let x_goog_request_params = [""; 0].into_iter().fold(String::new(), |b, p| b + "&" + &p);
+        let x_goog_request_params = {
+            use gaxi::routing_parameter::Segment;
+            gaxi::routing_parameter::format(&[gaxi::routing_parameter::value(
+                Some(&req).map(|v| v.upload_id.as_str()),
+                &[],
+                &[
+                    Segment::Literal("projects"),
+                    Segment::Literal("/"),
+                    Segment::SingleWildcard,
+                    Segment::Literal("/"),
+                    Segment::Literal("buckets"),
+                    Segment::Literal("/"),
```

Comment:
super nit: the `routing_parameter::value` tests had this as a single segment: `Literal("/buckets/")`. If you care enough, you could make the tests look more like the generated code.

Reply:
You are right, I meant to add a feature request to optimize this and I forgot: #1919.